llvm/docs/HowToCrossCompileLLVM.rst

```diff
@@ -6,210 +6,202 @@ Introduction
+
+    $ qemu-aarch64-static -L $SYSROOT ./build/aarch64-linux-gnu/bin/clang --version
+    clang version 21.0.0git (https://github.com/llvm/llvm-project cedfdc6e889c5c614a953ed1f44bcb45a405f8da)
+    Target: aarch64-unknown-linux-gnu
+    Thread model: posix
+    InstalledDir: /home/asb/llvm-project/build/aarch64-linux-gnu/bin
+
+    $ export QEMU_LD_PREFIX=$SYSROOT; ./build/aarch64-linux-gnu/bin/clang --version
+    clang version 21.0.0git (https://github.com/llvm/llvm-project cedfdc6e889c5c614a953ed1f44bcb45a405f8da)
+    Target: aarch64-unknown-linux-gnu
+    Thread model: posix
+    InstalledDir: /home/asb/llvm-project/build/aarch64-linux-gnu/bin
+
+Working around a ninja dependency issue
+---------------------------------------
+
+If you followed the instructions above to create a sysroot, you may run into a
+`longstanding problem related to path canonicalization in ninja
+<https://github.com/ninja-build/ninja/issues/1330>_`. GCC canonicalizes system
+headers in dependency files, so when ninja reads them it doesn't need to do
+so. Clang doesn't, and unfortunately ninja doesn't implement the
+canonicalization logic at all, meaning for some system headers with symlinks
+in the paths, it can incorrectly compute a non-existing path and consider it
+as always dirty. If you are suffering from this issue, you will find at any
+attempt at an incremental build (including the suggested command to build the
+``install`` target in the next section) results in recompiling everything.
+``ninja -C build/$TARGET -t deps`` shows files in ``$SYSROOT/include/*`` that
+don't (as the ``$SYSROOT/include`` folder doesn't exist) and you can
+further confirm these files are causing the build to be seen as dirty with
+``ninja -C build/$TARGET -d deps``. Until such time as the issue is avoided
+due to a change in behaviour for Clang or for Ninja, a functional workaround
+is to create a symlink so that the incorrect ``$SYSROOT/include/*``
+dependencies resolve to files within ``$SYSROOT/usr/include/*``. This works in
+practice for the simple cross-compilation use case described here, but isn't
+a general solution.
 
-It should automatically find out how many cores you have, what are
-the rules that needs building and will build the whole thing.
+   .. code-block:: bash
 
-You can't run ``ninja check-all`` on this tree because the created
-binaries are targeted to ARM, not x86_64.
+    sudo ln -s usr/include $SYSROOT/include
 
-Installing and Using
+Installing and using
 --------------------
 
-After the LLVM/Clang has built successfully, you should install it
-via:
-
-   .. code-block:: bash
-
-     $ ninja install
-
-which will create a sysroot on the install-dir. You can then tar
-that directory into a binary with the full triple name (for easy
-identification), like:
+After LLVM/Clang has built successfully, you can install it via:
```

Comment:
If we are going to say that most people do not need to set the install prefix setting in the previous section, it seems weird to show use of the install target here with no warning.

Maybe we should recommend that folks always set that setting to some path, just in case?

Removing a bunch of files scattered around my system is going to be a pain. It's ok for us working on imaged corporate machines or containers, but worse for anyone else.

Reply:
Maybe `..note: Use of the ``install`` target requires that you have <set the thing> otherwise <bad stuff happens>`.